engine/src/engine.ts

```diff
@@ -271,30 +271,22 @@ export default class Engine {
   // Tells the UI if the new move should be on the same line or not
   newTurn = true;
 
-  constructor(
-    moves: string[] = [],
-    options: EngineOptions = {},
-    engineVersion?: string,
-    replay?: boolean,
-    variantBoards?: Map<Faction, FactionBoardVariant>
-  ) {
+  constructor(moves: string[] = [], options: EngineOptions = {}, engineVersion?: string, replay?: boolean) {
     this.options = options;
     if (engineVersion) {
       this.version = engineVersion;
     }
     this.replay = replay;
-    if (replay) {
-      this.options.noFedCheck = true;
-      this.options.flexibleFederations = true;
+    if (this.options.factionVariantVersion === undefined) {
+      this.options.factionVariantVersion = latestVariantVersion(this.options.factionVariant);
```

Comment:
this seems wrong - the latest version of the beta variant is not the same for all factions.

Maybe it's OK, because we're loading it with `v.version <= customization.version` - we just have to make sure that any variant changes will use "3" for beta, even if this faction did not have any changes in version "2"

Reply:
I did this because I saw some factions with "version: 2" and no prior version, so I assumed the version was shared across factions

Reply:
Yes it actually is - I was more thinking about ways to make sure this is always the case.

Reply:
Well - the only trouble that a version desync can cause is if the version is updated after the game is started and a player chose their faction, the player might be on a specific version of the beta variant and the next player that chose after the update on a different version.

During replays, ... it shouldn't cause problems because what's used in priority is the faction stored in `player.variant.board`

Reply:
Good point  - we should fix the version when the game starts